src/DefinitionResolver.php

```diff
@@ -108,37 +166,60 @@ public function getDocumentationFromNode(Node $node)
      * @param string $fqn
      * @return Definition
      */
-    public function createDefinitionFromNode(Node $node, string $fqn = null): Definition
+    public function createDefinitionFromNode($node, string $fqn = null): Definition
     {
-        $parent = $node->getAttribute('parentNode');
         $def = new Definition;
-        $def->canBeInstantiated = $node instanceof Node\Stmt\Class_;
+        $def->fqn = $fqn;
+
+        // Determines whether the suggestion will show after "new"
+        $def->canBeInstantiated = $node instanceof Node\Statement\ClassDeclaration;
+
+        // Interfaces, classes, traits, namespaces, functions, and global const elements
         $def->isGlobal = (
-            $node instanceof Node\Stmt\ClassLike
-            || ($node instanceof Node\Name && $parent instanceof Node\Stmt\Namespace_)
-            || $node instanceof Node\Stmt\Function_
-            || $parent instanceof Node\Stmt\Const_
+            $node instanceof Node\Statement\InterfaceDeclaration ||
+            $node instanceof Node\Statement\ClassDeclaration ||
+            $node instanceof Node\Statement\TraitDeclaration ||
+
+            ($node instanceof Node\Statement\NamespaceDefinition && $node->name !== null) ||
+
+            $node instanceof Node\Statement\FunctionDeclaration ||
+
+            ($node instanceof Node\ConstElement && $node->parent->parent instanceof Node\Statement\ConstDeclaration)
         );
+
+        // Static methods and static property declarations
         $def->isStatic = (
-            ($node instanceof Node\Stmt\ClassMethod && $node->isStatic())
-            || ($node instanceof Node\Stmt\PropertyProperty && $parent->isStatic())
+            ($node instanceof Node\MethodDeclaration && $node->isStatic()) ||
+
+            (($propertyDeclaration = ParserHelpers::tryGetPropertyDeclaration($node)) !== null
+            && $propertyDeclaration->isStatic())
         );
-        $def->fqn = $fqn;
-        if ($node instanceof Node\Stmt\Class_) {
-            $def->extends = [];
-            if ($node->extends) {
-                $def->extends[] = (string)$node->extends;
-            }
-        } else if ($node instanceof Node\Stmt\Interface_) {
+
+        if ($node instanceof Node\Statement\ClassDeclaration &&
+            // TODO - this should be bette rrpreented in the parser API
+            $node->classBaseClause !== null && $node->classBaseClause->baseClass !== null) {
+            $def->extends = [(string)$node->classBaseClause->baseClass->getResolvedName()];
+            // TODO - why is this represented as an array?
+            // TODO interface implementations.
```

Comment:
It works, but then it doesn't match the old behavior. Then if I have `class C implements I`, and `I` has some methods that `C` implements, then they show up twice in completions. Unless we dedupe completions somewhere, it kind of seems right to ignore the implemented interfaces.

The current behavior seems to match that of the old parser. Side note, if a class overrides a base class' method, then that shows up twice, for probably the same reason.